File: Algorithm_problem_solving/Programers/Brute_Force/Mock_test/mock_test.py
import sys

input = sys.stdin.readline

# Retruns the number of points.
# param(correct answer, individual check way)
def abandonment(answers, check):
    lenAnswers = len(answers)
    select = 0

    for i in range(lenAnswers):
        if(answers[i] == check[i % len(check)]):
            select = select + 1

    return select


def solution(answers):
    answer = []

    first = (1, 2, 3, 4, 5)
    second = (2, 1, 2, 3, 2, 4, 2, 5)
    third = (3, 3, 1, 1, 2, 2, 4, 4, 5, 5)

    # Obtain each score
    score = []
    score.append(abandonment(answers, first))
    score.append(abandonment(answers, second))
    score.append(abandonment(answers, third))

    # Find out who's best
    maximum = max(score)
    for i in range(len(score)):
        if(score[i] == maximum):
            answer.append(i+1)

    return answer


if __name__ == "__main__":
    print(solution([1, 2, 3, 4, 5]))

File: Algorithm_problem_solving/Programers/Brute_Force/Mock_test/mock_test.js
function solution(answers) {
    var answer = [];
    var first_person = [1, 2, 3, 4, 5];
    var second_person = [2, 1, 2, 3, 2, 4, 2, 5];
    var third_person = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    var points = [0, 0, 0];

    for (var i = 0; i < answers.length; i++) {
        if (answers[i] == first_person[i % first_person.length])
            points[0]++;
        if (answers[i] == second_person[i % second_person.length])
            points[1]++;
        if (answers[i] == third_person[i % third_person.length])
            points[2]++;
    }

    // in js function for compare with list
    var maxnum = Math.max.apply(null, points);
    for (var i = 0; i < points.length; i++) {
        if (maxnum == points[i]) {
            answer.push(i + 1);
        }
    }

    return answer;
}